rust-connectors/sources/http/tests/get-test-full.bats

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bats
+
+setup() {
+    cargo build -p http-json-mock
+    ../../../target/debug/http-json-mock & disown
+    MOCK_PID=$!
+    FILE=$(mktemp --suffix .yaml)
+    cp ./tests/get-test-full-config.yaml $FILE
+    UUID=$(uuidgen)
+    TOPIC=${UUID}-topic
+
+    sed -i.BAK "s/http-json-connector/${UUID}/g" $FILE
+    IP_ADDRESS=$(ip route get 8.8.8.8 | awk -F"src " 'NR==1{split($2,a," ");print a[1]}')
+    sed -i.BAK "s/IP_ADDRESS/${IP_ADDRESS}/g" $FILE
+    fluvio connector create --config $FILE
+}
+
+teardown() {
+    fluvio connector delete $UUID
+    fluvio topic delete $TOPIC
+    kill $MOCK_PID
+}
+
+@test "http-connector-get-full-test" {
+    count=1
+    echo "Starting consumer on topic $TOPIC"
+    sleep 13
+
+    fluvio consume -o 0 -d $TOPIC | while read input; do
+        expected="Hello, Fluvio! - $count"
```

Comment:
This is what I initially thought too... but then I ran the test doing no changes and it passed.

And when I did something to make it break it wasn't very informative. I need to dig into the source of it what is going on.

Reply:
Maybe when we run the tests we are using the latest image from docker hub instead of the image with the changes from this PR. We need to check that

Reply:
Yeah. I think that might be the case. Right now, I think the fix for `version: dev` is in master but un-published.

Reply:
But on the other hand, I think that if were using the old image it would just fail due to the `output_format` configuration.

Reply:
I added get_full_requeest - now just need to find where to plumb it

utils/mocks/http-json-mock/src/main.rs
```
async fn get_full_request(req: Request<State>) -> tide::Result {
    let state = req.state();
    let value = state.get_count.fetch_add(1, Ordering::Relaxed) + 1;
    let body = format!("Hello, Fluvio! - {}", value);
    Ok(format!("HTTP/1.1 200 OK\ncontent-length: {}\n\n{}", body.length(), body).into())
}

```

Reply:
yaml is idiomatic foo_bar but CLI is just foo-bar because CLI idiomatic.. it's a Rust thing :woman_shrugging:

Reply:
(you hit this with a lot of crates as many of them are named with - but you have to use them via _)

Reply:
The problem we have, is that that the key,value that we have under `parameters` in the yaml config is used by the connector controller. [See](https://github.com/infinyon/fluvio/blob/e655710bcede6933d021b697b048dd8025d1c47b/crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs#L225)

So, if we use `output_format` in the yaml config, with the current fluvio version it will fail because it will call this with --output_format. We can either change [this](https://github.com/infinyon/fluvio/blob/e655710bcede6933d021b697b048dd8025d1c47b/crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs#L228) to replace all `_` in key with `-`. Or add support for both _ and - in the CLI. What do you think?

Reply:
I would think fluvio needs to support CLI idiomatic "-" automatically and translate _ to - due to:

I don't think I can use "-" as Rust requires "_"
Whilst we use StructOpt which does not support "_" via CLI.

Reply:
I raised a separate issue #134 here - I propose that we claim and leave this as identified technical debt.